meteor/client/ui/Settings/ShowStyle/VariantListItem.tsx

```diff
@@ -95,6 +95,80 @@ export const VariantListItem: React.FunctionComponent<IShowStyleVariantItemProps
 		})
 	}
 
+	function showStyleVariantNameAlreadyExists(): boolean {
+		return props.dragDropVariants.some((variant: ShowStyleVariant) => {
+			return variant.name === props.showStyleVariant.name && variant._id !== props.showStyleVariant._id
+		})
+	}
+
+	function addDuplicationCountToName(showStyleVariant: ShowStyleVariant, duplicatorIncrement?: number): void {
+		const increment = duplicatorIncrement ?? 0
+		showStyleVariant.name = getNameWithRemovedDuplicateIndicator(showStyleVariant)
+		showStyleVariant.name += ' (' + (getDuplicatedShowStyleVariantCount(showStyleVariant) + increment) + ')'
+		if (showStyleVariantNameAlreadyExists()) {
+			addDuplicationCountToName(showStyleVariant, increment + 1)
+		} else {
+			MeteorCall.showstyles.renameShowStyleVariant(props.showStyleBase._id, props.showStyleVariant).catch(logger.warn)
+		}
+	}
+
+	function getDuplicatedShowStyleVariantCount(showStyleVariant: ShowStyleVariant): number {
+		const variantsWithName = props.dragDropVariants.filter((variant: ShowStyleVariant) => {
+			const variantNameWithoutDuplicator = getNameWithRemovedDuplicateIndicator(variant)
+			return variantNameWithoutDuplicator === showStyleVariant.name && variant._id !== showStyleVariant._id
+		})
+		return variantsWithName.length
+	}
+
+	function getNameWithRemovedDuplicateIndicator(showStyleVariant: ShowStyleVariant): string {
+		return showStyleVariant.name.replace(/ +\(\d+\)$/, '')
+	}
+
+	function replaceOtherShowStyleVariant(): void {
+		const variantToReplace = props.dragDropVariants.find((variant: ShowStyleVariant) => {
+			return props.showStyleVariant.name === variant.name && props.showStyleVariant._id !== variant._id
+		})
+		if (!variantToReplace) {
+			return
+		}
+		MeteorCall.showstyles.removeShowStyleVariant(variantToReplace._id).catch(logger.warn)
+	}
+
+	function provideDuplicatedNameOptions() {
+		doModalDialog({
+			title: t('Two variants have the same name, do you want to keep both?'),
+			yes: t('Keep both'),
+			no: t('Replace'),
+			onAccept: () => {
+				addDuplicationCountToName(props.showStyleVariant)
+				removeAllQueueItems()
+			},
+			onSecondary: () => {
+				replaceOtherShowStyleVariant()
+				removeAllQueueItems()
+			},
+			onDiscard: () => {
+				addDuplicationCountToName(props.showStyleVariant)
+				removeAllQueueItems()
+			},
+			message: (
+				<React.Fragment>
+					<p>
+						{t('Do you want to replace "{{showStyleVariantName}}"?', {
```

Comment:
I was greeted with this message when opening the Show Style Variants page, But I had more than 5 variants with the same name. In this scenario it is hard to tell which will replace which. I think this message could be more specific.

Reply:
That's a good point. I think your comments arise from the fact that i've had some issues with what information is exposed and available from where i have to check for duplication. 
Specifically for the message, i'm not sure what to put more than the name, since the user would not know any distinguishing factors other than that. I'll look into it though, but if you have any ideas please let me know!

Reply:
I see the weird behaviour when copying arise from the fact that the useEffect has a dependency on the variant name, i did not consider that it would cause issues in the copying-scenario. I will find a fix for that!